### DISCUSSION TITLE
use(Path(..)) could work on an iterator like Path.glob or a list of paths?

### DISCUSSION THREAD

**Discussion by amogorkon** [amogorkon/justuse#309] [Ideas]
Thinking about Path, could it be useful to accept an iterator of path and return an iterator of modules?

**Comment by greyblue9** [comment]
What about a dict of modules? Then the user could do

```python
mod_dict = use(paths)
globals().update(mod_dict)
```
etc.

Of course, that idea assumes we can determine the modules' names.

**Comment by amogorkon** [comment]
Okay, that certainly might be something we could do, do you have an example where this might be actually useful?

**Comment by greyblue9** [comment]
My thought was, I have a utility function called lsmod(mod) that attempts to discover all submodules of the module you pass in, as a list of names. If I change this to return a list of paths, it sounds like it would work as input to use() in this scenario.

I currently use another utility function, importAll(*names),  to import all the named modules, then I can easily see what the modules names are with corresponding module values.

I guess in this case, we can just go with the simpler option of returning a list, though?
